bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/handler/SunSynkInverterHandler.java

```diff
@@ -0,0 +1,416 @@
+                case CHANNEL_BATTERY_INTERVAL_2_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            2);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_3_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            3);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_4_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            4);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_5_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            5);
+                    break;
+                case CHANNEL_BATTERY_INTERVAL_6_POWER_LIMIT:
+                    this.tempInverterChargeSettings.setIntervalBatteryPowerLimit(Integer.valueOf(command.toString()),
+                            6);
+                    break;
+                // Inverter control
+                case CHANNEL_INVERTER_CONTROL_TIMER:
+                    this.tempInverterChargeSettings.setPeakAndValley(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_ENERGY_PATTERN:
+                    this.tempInverterChargeSettings.setEnergyMode(Integer.valueOf(command.toString()));
+                    break;
+                case CHANNEL_INVERTER_CONTROL_WORK_MODE:
+                    this.tempInverterChargeSettings.setSysWorkMode(Integer.valueOf(command.toString()));
+                    break;
+            }
+
+            this.batterySettingsUpdated = true; // true = update battery settings from API at next interval
+        }
+    }
+
+    private void sendSettingsCommandToInverter() {
+        logger.debug("Ok - will handle command for CHANNEL_BATTERY_INTERVAL_1_GRID_CHARGE");
+        try {
+            inverter.sendSettings(this.tempInverterChargeSettings);
+        } catch (SunSynkSendCommandException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Could not send command to inverter " + config.getAlias() + ". Authentication Failure !");
+            return;
+        }
+        logger.debug("Sent command: to inverter {}.", config.getAlias());
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Running dispose()");
+        ScheduledFuture<?> refreshTask = this.refreshTask;
+        if (refreshTask != null) {
+            refreshTask.cancel(true);
+            this.refreshTask = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        logger.debug("Will boot up the inverter binding!");
```

Comment:
don't do this, the framework will log these messages
```suggestion
```

Reply:
okay